If decommission was aborted and DecommissionNemesis failed, need to check node was decommissioned and clean db_cluster.nodes

### Prerequisites

* [x] Are you rebased to master ?
* [ ] Is it reproducible ?
* [ ] Did you perform a cursory search if this issue isn't opened ?

### Versions
* **SCT**: master
* **scylla**: master

### Logs
* test_id: 96a7d2d0-91a1-45a6-a853-2e383a82cb1d
* job log: https://cloudius-jenkins-test.s3.amazonaws.com/96a7d2d0-91a1-45a6-a853-2e383a82cb1d/20230826_072309/sct-96a7d2d0.log.tar.gz
* Argus: https://argus.scylladb.com/test/024e78ba-3671-4fd0-bafc-b48b2a978f59/runs?additionalRuns[]=96a7d2d0-91a1-45a6-a853-2e383a82cb1d

### Description

During job DecommissionNemeis was running. It is failed due to nerwork error. But decommission was continued in background and finished. But because of DecommissionNemesis was aborted, node9 stayed in db_cluster.nodes. So when this node9 was chosen  by another nemesis and run nodetool status, result of nodetool status was wrong and nemesises are failed to run:

Result of nodetool status on node10:
```
< t:2023-08-26 06:29:48,294 f:base.py         l:228  c:RemoteLibSSH2CmdRunner p:DEBUG > Datacenter: eu-west
< t:2023-08-26 06:29:48,294 f:base.py         l:228  c:RemoteLibSSH2CmdRunner p:DEBUG > ===================
< t:2023-08-26 06:29:48,294 f:base.py         l:228  c:RemoteLibSSH2CmdRunner p:DEBUG > Status=Up/Down
< t:2023-08-26 06:29:48,294 f:base.py         l:228  c:RemoteLibSSH2CmdRunner p:DEBUG > |/ State=Normal/Leaving/Joining/Moving
< t:2023-08-26 06:29:48,294 f:base.py         l:228  c:RemoteLibSSH2CmdRunner p:DEBUG > --  Address     Load       Tokens       Owns    Host ID                               Rack
< t:2023-08-26 06:29:48,298 f:base.py         l:228  c:RemoteLibSSH2CmdRunner p:DEBUG > UN  10.4.10.3   32.12 GB   256          ?       13223bab-08eb-4667-b255-bbb6a82528e1  1c
< t:2023-08-26 06:29:48,300 f:base.py         l:228  c:RemoteLibSSH2CmdRunner p:DEBUG > UN  10.4.8.115  ?          256          ?       77a67007-fe40-4e0c-a553-33284fb2ae35  1c
< t:2023-08-26 06:29:48,306 f:base.py         l:228  c:RemoteLibSSH2CmdRunner p:DEBUG > UN  10.4.10.13  29.32 GB   256          ?       ae82318c-3d44-4dd5-a09c-365f5325c5a0  1c
< t:2023-08-26 06:29:48,306 f:base.py         l:228  c:RemoteLibSSH2CmdRunner p:DEBUG > UN  10.4.8.171  32.21 GB   256          ?       cebd69e7-ddd6-45df-be94-d72a94f0e754  1c
< t:2023-08-26 06:29:48,306 f:base.py         l:228  c:RemoteLibSSH2CmdRunner p:DEBUG > Datacenter: eu-west_nemesis_dc
< t:2023-08-26 06:29:48,306 f:base.py         l:228  c:RemoteLibSSH2CmdRunner p:DEBUG > ==============================
< t:2023-08-26 06:29:48,306 f:base.py         l:228  c:RemoteLibSSH2CmdRunner p:DEBUG > Status=Up/Down
< t:2023-08-26 06:29:48,306 f:base.py         l:228  c:RemoteLibSSH2CmdRunner p:DEBUG > |/ State=Normal/Leaving/Joining/Moving
< t:2023-08-26 06:29:48,306 f:base.py         l:228  c:RemoteLibSSH2CmdRunner p:DEBUG > --  Address     Load       Tokens       Owns    Host ID                               Rack
< t:2023-08-26 06:29:48,308 f:base.py         l:228  c:RemoteLibSSH2CmdRunner p:DEBUG > UN  10.4.11.30  1.43 MB    256          ?       57804c9f-c18f-4dff-a5d8-cde078717963  1c

```

Result of node9:
```
 t:2023-08-26 06:30:14,360 f:base.py         l:228  c:RemoteLibSSH2CmdRunner p:DEBUG > Datacenter: eu-west
< t:2023-08-26 06:30:14,360 f:base.py         l:228  c:RemoteLibSSH2CmdRunner p:DEBUG > ===================
< t:2023-08-26 06:30:14,360 f:base.py         l:228  c:RemoteLibSSH2CmdRunner p:DEBUG > Status=Up/Down
< t:2023-08-26 06:30:14,360 f:base.py         l:228  c:RemoteLibSSH2CmdRunner p:DEBUG > |/ State=Normal/Leaving/Joining/Moving
< t:2023-08-26 06:30:14,364 f:base.py         l:228  c:RemoteLibSSH2CmdRunner p:DEBUG > --  Address     Load       Tokens       Owns    Host ID                               Rack
< t:2023-08-26 06:30:14,364 f:base.py         l:228  c:RemoteLibSSH2CmdRunner p:DEBUG > UN  10.4.10.3   32.46 GB   256          ?       13223bab-08eb-4667-b255-bbb6a82528e1  1c
< t:2023-08-26 06:30:14,366 f:base.py         l:228  c:RemoteLibSSH2CmdRunner p:DEBUG > UN  10.4.8.115  30.73 GB   256          ?       77a67007-fe40-4e0c-a553-33284fb2ae35  1c
< t:2023-08-26 06:30:14,367 f:base.py         l:228  c:RemoteLibSSH2CmdRunner p:DEBUG > UN  10.4.10.13  31.74 GB   256          ?       ae82318c-3d44-4dd5-a09c-365f5325c5a0  1c
< t:2023-08-26 06:30:14,369 f:base.py         l:228  c:RemoteLibSSH2CmdRunner p:DEBUG > UN  10.4.8.171  33.93 GB   256          ?       cebd69e7-ddd6-45df-be94-d72a94f0e754  1c
< t:2023-08-26 06:30:14,369 f:base.py         l:228  c:RemoteLibSSH2CmdRunner p:DEBUG > 

```
This result failed another nemesis: such as 
disrupt_add_remove_dc with error: AssertionError: new datacenter was not registered
disrupt_bootstrap_streaming_error: sdcm.cluster.ClusterNodesNotReady: Not all nodes joined the cluster


### Steps to Reproduce

1. [First Step]
2. [Second Step]
3. [and so on...]

**Expected behavior:** [What you expected to happen]

**Actual behavior:** [What actually happened]

--- a/sdcm/nemesis.py
+++ b/sdcm/nemesis.py
@@ -4914,7 +4914,7 @@ class Nemesis:  # pylint: disable=too-many-instance-attributes,too-many-public-m
 
         bootstrapabortmanager = NodeBootstrapAbortManager(bootstrap_node=new_node, verification_node=self.target_node)
 
-        bootstrapabortmanager.run_bootsrap_and_abort_with_action(terminate_pattern, abort_action=new_node.stop_scylla)
+        bootstrapabortmanager.run_bootstrap_and_abort_with_action(terminate_pattern, abort_action=new_node.stop_scylla)
         bootstrapabortmanager.clean_and_restart_bootstrap_after_abort()
 
         if new_node.db_up() and not self.target_node.raft.is_cluster_topology_consistent():

--- a/sdcm/utils/raft/common.py
+++ b/sdcm/utils/raft/common.py
@@ -43,7 +43,8 @@ def validate_raft_on_nodes(nodes: list["BaseNode"]) -> None:
 
 
 class NodeBootstrapAbortManager:
-    INSTANCE_START_TIMEOUT = 300
+    INSTANCE_START_TIMEOUT = 600
+    SUCCESS_BOOTSTRAP_TIMEOUT = 3600
 
     def __init__(self, bootstrap_node: BaseNode, verification_node: BaseNode):
         self.bootstrap_node = bootstrap_node
@@ -60,18 +61,18 @@ class NodeBootstrapAbortManager:
             self.bootstrap_node.stop_wait_db_up_event.set()
         LOGGER.debug("Stop event was set for node %s", self.bootstrap_node.name)
 
-    def _start_bootstrap(self, timeout=3600):
+    def _start_bootstrap(self):
         try:
             LOGGER.debug("Starting bootstrap process %s", self.bootstrap_node.name)
-            self.bootstrap_node.parent_cluster.node_setup(self.bootstrap_node, verbose=True, timeout=timeout)
+            self.bootstrap_node.parent_cluster.node_setup(self.bootstrap_node, verbose=True)
+            LOGGER.debug("Node %s was bootstrapped", self.bootstrap_node.name)
         except Exception as exc:  # pylint: disable=broad-except
             LOGGER.error("Setup failed for node %s with err %s", self.bootstrap_node.name, exc)
         finally:
             self._set_wait_stop_event()
-        LOGGER.debug("Node %s was bootstrapped", self.bootstrap_node.name)
 
     def _abort_bootstrap(self, abort_action: Callable, log_message: str, timeout: int = 600):
-        LOGGER.debug("Stop bootsrap process after log message: '%s'", log_message)
+        LOGGER.debug("Stop bootstrap process after log message: '%s'", log_message)
         log_follower = self.bootstrap_node.follow_system_log(patterns=[log_message])
         try:
             wait_for(func=lambda: list(log_follower), step=5,
@@ -80,9 +81,9 @@ class NodeBootstrapAbortManager:
                      throw_exc=True,
                      stop_event=self.bootstrap_node.stop_wait_db_up_event)
             abort_action()
-            LOGGER.info("Scylla was stopped succesfully on node %s", self.bootstrap_node.name)
+            LOGGER.info("Scylla was stopped successfully on node %s", self.bootstrap_node.name)
         except Exception as exc:  # pylint: disable=broad-except
-            LOGGER.warning("Abort was failed on node %s with errof %s", self.bootstrap_node.name, exc)
+            LOGGER.warning("Abort was failed on node %s with error %s", self.bootstrap_node.name, exc)
         finally:
             self._set_wait_stop_event()
 
@@ -95,7 +96,7 @@ class NodeBootstrapAbortManager:
                 LOGGER.debug("Node %s has host id: %s in log", self.bootstrap_node.name, new_node_host_id)
                 node_host_ids.append(new_node_host_id)
         self.bootstrap_node.log.debug("New host was not properly bootstrapped. Terminate it")
-        self.db_cluster.terminate_node(self)
+        self.db_cluster.terminate_node(self.bootstrap_node)
         self.monitors.reconfigure_scylla_monitoring()
         self.verification_node.raft.clean_group0_garbage(raise_exception=True)
         if node_host_ids:
@@ -105,25 +106,24 @@ class NodeBootstrapAbortManager:
 
         assert self.verification_node.raft.is_cluster_topology_consistent(), \
             "Group0, Token Ring and number of node in cluster are differs. Check logs"
-        self.bootstrap_node.parent_cluster.check_nodes_up_and_normal()
-        LOGGER.info("Failed bootstrapped node %s removed. Cluster is in initial state", self.bootstrap_node.name)
+        self.verification_node.parent_cluster.check_nodes_up_and_normal()
+        LOGGER.info("Failed bootstrapped node %s was removed. Cluster is in initial state", self.bootstrap_node.name)
 
-    def run_bootsrap_and_abort_with_action(self, terminate_pattern, abort_action: Callable):
-        trigger = partial(
-            self._start_bootstrap)
-
+    def run_bootstrap_and_abort_with_action(self, terminate_pattern, abort_action: Callable, abort_action_timeout: int = 300):
         watcher = partial(self._abort_bootstrap,
                           abort_action=abort_action,
                           log_message=terminate_pattern.log_message,
-                          timeout=terminate_pattern.timeout + self.INSTANCE_START_TIMEOUT)
+                          timeout=self.INSTANCE_START_TIMEOUT + terminate_pattern.timeout + abort_action_timeout)
 
+        wait_operations_timeout = (self.SUCCESS_BOOTSTRAP_TIMEOUT + self.INSTANCE_START_TIMEOUT
+                                   + terminate_pattern.timeout + abort_action_timeout)
         with ignore_stream_mutation_fragments_errors(), contextlib.ExitStack() as stack:
             for expected_start_failed_context in self.verification_node.raft.get_severity_change_filters_scylla_start_failed(
                     terminate_pattern.timeout):
                 stack.enter_context(expected_start_failed_context)
             try:
-                ParallelObject(objects=[trigger, watcher],
-                               timeout=terminate_pattern.timeout + self.INSTANCE_START_TIMEOUT).call_objects(ignore_exceptions=True)
+                ParallelObject(objects=[self._start_bootstrap, watcher],
+                               timeout=wait_operations_timeout).call_objects(ignore_exceptions=True)
             finally:
                 self._set_wait_stop_event()
 

